lib/ImageData.js

```diff
@@ -161,7 +161,12 @@ class ImageData {
         const suffix = fileSuffix || "";
         const fileName = path.parse(this.baseName).name;
         const extension = "." + this.type.ext;
-        if ( directory != null ) {
+
+        if ( typeof directory === "object" && directory.regex && directory.regex.find ) {
+            const regex = new RegExp(directory.regex.find);
+            const output = this.dirName.replace(regex, directory.regex.replace || "");
+            return path.join(output, prefix + fileName + suffix + extension);
```

Comment:
Sure, I agree with complexity part, at least partially. It's because I think that better flexibility always will introduce more complexity, in one or different form. Also, keep in mind that we are not replacing old simple approach, we are just adding extension to it. So the `RegEx` configuration could be used only by more advanced users. But I guess I must agree, that this could increase number of incoming questions and bugs :)

If you feel that the [path-template](https://github.com/matsadler/path-template) approach is simpler, we can start from there.